src/sidebar/components/AnnotationView.tsx

```diff
@@ -56,9 +59,7 @@ function AnnotationView({ loadAnnotationsService, onLogin }) {
 
         // Make the full thread of annotations visible. By default replies are
         // not shown until the user expands the thread.
-        annots.forEach(annot =>
-          store.setExpanded(/** @type {string} */ (annot.id), true)
-        );
+        annots.forEach(annot => annot.id && store.setExpanded(annot.id, true));
```

Comment:
I'm not sure what's the best way to do this. Previously we seemed to be neglecting the fact that `annot.id` can be undefined by doing an explicit casting.

Other possible approaches are:

```ts
if (annot.id) {
  store.setExpanded(annot.id, true)
}
```

```ts
store.setExpanded(annot.id ?? '', true)
```

```ts
store.setExpanded(annot.id!, true)
```

Reply:
Ah, OK, interesting. `id` on `Annotation` is, as you noted, is `string|undefined`. This is because the `id` is owned by the service, so annotations created in the app but not yet saved will not have an `id` (every annotation in the `client` app _does_ have a unique identifier at all times; that's what `$tag` is for).

In this case, we've just loaded annotations from the service, so in general we are guaranteed that every annotation has an `id`. I think the way you've handled it here keeps us out of unpreventable trouble and seems like a fine approach. You _could_ use `!` but this value is not "ours" and so I generally prefer your variant.